src/sparseml/pytorch/torchvision/train.py

```diff
@@ -95,14 +95,19 @@ def train_one_epoch(
         metric_logger.meters["img/s"].update(batch_size / (time.time() - start_time))
 
 
-def evaluate(model, criterion, data_loader, device, print_freq=100, log_suffix=""):
+def evaluate(
+    model, criterion, data_loader, device, args, print_freq=100, log_suffix=""
+):
     model.eval()
     metric_logger = utils.MetricLogger(delimiter="  ")
     header = f"Test: {log_suffix}"
 
     num_processed_samples = 0
     with torch.inference_mode():
-        for image, target in metric_logger.log_every(data_loader, print_freq, header):
+        for i, (image, target) in zip(
+            range(args.max_eval_steps),
+            metric_logger.log_every(data_loader, print_freq, header),
```

Comment:
why not just `enumerate`?

Reply:
the `zip(range(), ...)` makes the loop stop when the range runs out (ie when max_eval_steps is reached). its kinda like doing a slice on an interable (`data_loader[:max_eval_steps]`). maybe its too subtle?

Reply:
Updated logic

Reply:
args.max_eval_steps is always set and will be an int, I feel this idiom is very well accepted in python community, and it also works if the second iterator is smaller than the first.

A nit I have is to rename `i` to `_` because it's a throw away variable